.builder/actions/build_samples.py

```diff
@@ -5,9 +5,14 @@
 
 class BuildSamples(Builder.Action):
     def run(self, env):
+        skip_da = False
+        skip_samples = False
+        if env.args.cli_config['variables'].get('skip_da', "0") != "0":
```

Comment:
If possible, I feel like these should be passed as flags (--skip_da) through the builder and into here.

In fact, why do this at all?  Why not always build the samples?  Running them is separate and based on your variant.

Reply:
We don't always build the samples because the BYO_CRYPTO mode. With BYO_CRYPTO, the script will not build the tls implementation, and the builder could not build any test sample with it. 

I think you are right, I can change it to use variant.